### PR TITLE
Game Collection Fixes

### LINKED ISSUE #7051: Games on multi-game discs fail to boot with updates/game data installed
As a continuation of #4448, in pr #4450 we got basic support for multi-game discs. And it works fine until you install the game data or update, then it will fail to boot.

Log excerpt from the failed boot:

```
·! 0:00:13.339912 LDR: Disc: E:/FakeCDs/PS3 Games/[BLES 00635] TEKKEN 6/
·! 0:00:13.339915 LDR: Game: E:/FakeCDs/PS3 Games/[BLES 00635] TEKKEN 6/PS3_GM02/
·S 0:00:13.341529 LDR: Updates found at /dev_hdd0/game/BLES01250/!
·! 0:00:13.350362 LDR: Path: C:/Games/Emulators Collection/Emulators/PS3/rpcs3/dev_hdd0/game/BLES01250/USRDIR/EBOOT.BIN
·! 0:00:13.351489 LDR: Disc: E:/FakeCDs/PS3 Games/[BLES 00635] TEKKEN 6/
·! 0:00:13.351491 LDR: Game: E:/FakeCDs/PS3 Games/[BLES 00635] TEKKEN 6/PS3_GAME/
·E 0:00:13.351636 LDR: Unexpected disc directory for the disc game BLES01250 (found BLES00635)
```

[RPCS3.sc5_no_updates.log.gz](https://github.com/RPCS3/rpcs3/files/3913272/RPCS3.sc5_no_updates.log.gz)
[RPCS3.sc5_with_updates.log.gz](https://github.com/RPCS3/rpcs3/files/3913273/RPCS3.sc5_with_updates.log.gz)

### LINKED ISSUE #7544: Games on multi-game discs have duplicate entries in the library
## Quick summary
This disc has three games on it, and I get triple entries for each
library duplicates

## Details
Regression in build 9591 (pr #7481). I think some of it was addressed in #7488, but not for the multi-game collections.

### PR DESCRIPTION
**Pull request by Megamouse**
1. Do not reset PS3_GAME when booting disc patches
2. Skip some unnecessary code and logging during add_only
3. Fix duplicate paths in game list
4. Simplify some QtConcurrent code

fixes #7051
fixes #7544

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -987,7 +991,7 @@ void Emulator::Load(const std::string& title_id, bool add_only, bool force_globa
 		const std::string hdd0_disc = vfs::get("/dev_hdd0/disc/");
 		const std::size_t game_dir_size = 8; // size of PS3_GAME and PS3_GMXX
 		const std::size_t bdvd_pos = m_cat == "DG" && bdvd_dir.empty() && disc.empty() ? elf_dir.rfind("/USRDIR") - game_dir_size : 0;
-		const bool from_hdd0_game = m_path.find(hdd0_game) != umax;
+		const bool from_hdd0_game = m_path.find(hdd0_game) != std::string::npos;
```
Comment (Nekotekina) [inline rpcs3/Emu/System.cpp]: What's the problem with umax?

Comment (Megamouse) [reply rpcs3/Emu/System.cpp]: why use umax if you can use the standard value that's proposed in official examples?

Comment (Nekotekina) [reply rpcs3/Emu/System.cpp]: It's too long

Comment (Megamouse) [reply rpcs3/Emu/System.cpp]: to me this is easier to understand than some umax on a string find

**Comment by 13xforever** [comment]
Library looks good now
image

**Pull request merged**